src/main/scala/com/fulcrumgenomics/bam/FilterBam.scala

```diff
@@ -51,13 +52,13 @@ import scala.collection.JavaConversions._
   "read 2 is included, but not both), but does not update any flag fields.",
   group = classOf[SamOrBam])
 class FilterBam
-(@arg(doc = "If supplied, remove all reads that do not overlap the provided intervals.") var intervals: Option[Path] = None,
- @arg(doc = "Input BAM file.")                                         var input: Path,
- @arg(doc = "Output BAM file.")                                        var output: Path,
- @arg(doc = "If true remove all reads that are marked as duplicates.") var removeDuplicates: Boolean = true,
- @arg(doc = "Remove all unmapped reads.")                              var removeUnmappedReads: Boolean = true,
- @arg(doc = "Remove all reads with MAPQ lower than this number.")      var minMapQ: Int = 1,
- @arg(doc = "Remove all reads marked as secondary alignments.")        var removeSecondaryAlignments: Boolean = true
+( @arg(doc = "If supplied, remove all reads that do not overlap the provided intervals.") var intervals: Option[PathToIntervals] = None,
+  @arg(doc = "Input BAM file.")                                         var input: PathToBam,
+  @arg(doc = "Output BAM file.")                                        var output: PathToBam,
+  @arg(doc = "If true remove all reads that are marked as duplicates.") var removeDuplicates: Boolean = true,
+  @arg(doc = "Remove all unmapped reads.")                              var removeUnmappedReads: Boolean = true,
+  @arg(doc = "Remove all reads with MAPQ lower than this number.")      var minMapQ: Int = 1,
+  @arg(doc = "Remove all reads marked as secondary alignments.")        var removeSecondaryAlignments: Boolean = true
```

Comment:
I can't comment below for some reason, but I'm wondering if the `iterator.filterNot.map.sum` really works iteratively, or if it ends up materializing a List/Array or millions of longs of value 1 prior to summing?